plugins/module_utils/network/sonic/config/lst/lst.py

```diff
@@ -0,0 +1,518 @@
+    get_diff,
+    remove_empties,
+    update_states
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.sonic import (
+    edit_config,
+    to_request
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.utils.formatted_diff_utils import (
+    __DELETE_CONFIG_IF_NO_SUBCONFIG,
+    get_new_config,
+    get_formatted_config_diff
+)
+
+LST_PATH = 'data/openconfig-lst-ext:lst'
+PATCH = 'patch'
+DELETE = 'delete'
+TEST_KEYS = [
+    {'lst_groups': {'name': ''}},
+    {'interfaces': {'name': ''}},
+    {'upstream_groups': {'group_name': ''}}
+]
+TEST_KEYS_generate_config = [
+    {'lst_groups': {'name': '', '__delete_op': __DELETE_CONFIG_IF_NO_SUBCONFIG}},
+    {'interfaces': {'name': '', '__delete_op': __DELETE_CONFIG_IF_NO_SUBCONFIG}},
+    {'upstream_groups': {'group_name': '', '__delete_op': __DELETE_CONFIG_IF_NO_SUBCONFIG}}
+]
+
+
+class Lst(ConfigBase):
+    """
+    The sonic_lst class
+    """
+
+    gather_subset = [
+        '!all',
+        '!min',
+    ]
+
+    gather_network_resources = [
+        'lst',
+    ]
+
+    def __init__(self, module):
+        super(Lst, self).__init__(module)
+
+    def get_lst_facts(self):
+        """
+        Get the 'facts' (the current configuration)
+        :rtype: A dictionary
+        :returns: The current configuration as a dictionary
+        """
+        facts, _warnings = Facts(self._module).get_facts(self.gather_subset, self.gather_network_resources)
+        lst_facts = facts['ansible_network_resources'].get('lst')
+        if not lst_facts:
+            return {}
+        return lst_facts
+
+    def execute_module(self):
+        """
+        Execute the module
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        warnings = []
+        commands = []
+
+        existing_lst_facts = self.get_lst_facts()
+        commands, requests = self.set_config(existing_lst_facts)
+        if commands and len(requests) > 0:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.code)
+            result['changed'] = True
+        result['commands'] = commands
+
+        changed_lst_facts = self.get_lst_facts()
+
+        result['before'] = existing_lst_facts
+        if result['changed']:
+            result['after'] = changed_lst_facts
+
+        new_config = changed_lst_facts
+        old_config = existing_lst_facts
+        if self._module.check_mode:
+            result.pop('after', None)
+            new_config = get_new_config(commands, existing_lst_facts, TEST_KEYS_generate_config)
+            self.post_process_generated_config(new_config)
+            result['after(generated)'] = new_config
```

Comment:
Facts retrieval after configuration change, can be avoided in check mode.
```suggestion
        old_config = existing_lst_facts
        if self._module.check_mode:
            new_config = get_new_config(commands, existing_lst_facts, TEST_KEYS_generate_config)
            self.post_process_generated_config(new_config)
            result['after(generated)'] = new_config
        else:
            new_config = self.get_lst_facts()
            if result['changed']:
                result['after'] = new_config
```